aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -819,21 +825,41 @@ module aptos_framework::stake {
         assert!(signer::address_of(account) == stake_pool.operator_address, error::invalid_argument(ENOT_OPERATOR));
 
         let validator_set = borrow_global_mut<ValidatorSet>(@aptos_framework);
-        // Validate that the validator is already part of the validator set.
-        let maybe_index = find_validator(&validator_set.active_validators, pool_address);
-        assert!(option::is_some(&maybe_index), error::invalid_argument(ENOT_VALIDATOR));
-        let index = option::extract(&mut maybe_index);
-
-        let validator_info = vector::swap_remove(&mut validator_set.active_validators, index);
-        assert!(vector::length(&validator_set.active_validators) > 0, error::invalid_argument(ELAST_VALIDATOR));
-        vector::push_back(&mut validator_set.pending_inactive, validator_info);
-
-        event::emit_event<LeaveValidatorSetEvent>(
-            &mut stake_pool.leave_validator_set_events,
-            LeaveValidatorSetEvent {
-                pool_address,
-            },
-        );
+        // If the validator is still pending_active, directly kick the validator out.
+        let maybe_pending_active_index = find_validator(&validator_set.pending_active, pool_address);
+        if (option::is_some(&maybe_pending_active_index)) {
+            vector::swap_remove(
+                &mut validator_set.pending_active, option::extract(&mut maybe_pending_active_index));
+
+            // Decrease the voting power increase as the pending validator's voting power was added when they requested
+            // to join. Now that they changed their mind, their voting power should not affect the joining limit of this
+            // epoch.
+            let validator_stake = (coin::value(&stake_pool.active) as u128);
```

Comment:
probably should use get_next_epoch_voting_power to be consistent with joining function, there's edge case of pending_inactive if it's unlocked

Reply:
Makes sense. Updated. Pending active validators could have pending_inactive stake if they left and want to rejoin the validator set.